### PR TITLE
Invert some logic to avoid nested brace hell

### PR DESCRIPTION
**Pull request by jduck**

### CODE REVIEW

File: libr/debug/trace.c [hunk 1]
```diff
@@ -50,24 +50,36 @@ R_API int r_debug_trace_pc (RDebug *dbg, ut64 pc) {
...
-					if (dbg->anal->trace) {
-						r_anal_esil_trace (dbg->anal->esil, &op);
-					}
-				}
-				r_debug_trace_add (dbg, oldpc, op.size);
+	r_debug_reg_sync (dbg, R_REG_TYPE_GPR, false);
+	ri = r_reg_get (dbg->reg, dbg->reg->name[R_REG_NAME_PC], -1);
+	if (!ri) {
+		eprintf ("trace_pc: cannot get program counter\n");
+		return false;
```
Comment (radare) [inline libr/debug/trace.c]: if you return there's no need for an else block

File: libr/debug/trace.c [hunk 2]
```diff
@@ -50,24 +50,36 @@ R_API int r_debug_trace_pc (RDebug *dbg, ut64 pc) {
...
+			r_debug_trace_add (dbg, oldpc, op.size);
 		}
-		else
-			eprintf ("trace_pc: cannot get opcode size at 0x%"PFMT64x"\n", pc);
+		oldpc = addr;
+		return true;
 	}
-	//else
-	//	eprintf ("trace_pc: cannot read memory at 0x%"PFMT64x"\n", pc);
 	return false;
```
Comment (radare) [inline libr/debug/trace.c]: dead code

File: libr/debug/trace.c [hunk 3]
```diff
@@ -50,24 +50,36 @@ R_API int r_debug_trace_pc (RDebug *dbg, ut64 pc) {
...
+	if (!ri) {
+		eprintf ("trace_pc: cannot get program counter\n");
+		return false;
+	} else {
+		ut64 addr = r_reg_get_value (dbg->reg, ri);
+		if (!addr) {
+			return false;
+		}
+
+		if (dbg->iob.read_at (dbg->iob.io, addr, buf, sizeof (buf)) <= 0) {
```
Comment (radare) [inline libr/debug/trace.c]: must be != sizeof (buf)

File: libr/debug/trace.c [hunk 4]
```diff
@@ -50,24 +50,36 @@ R_API int r_debug_trace_pc (RDebug *dbg, ut64 pc) {
...
+			//eprintf ("trace_pc: cannot read memory at 0x%"PFMT64x"\n", addr);
+			return false;
+		}
+
+		if (r_anal_op (dbg->anal, &op, addr, buf, sizeof (buf)) <= 0) {
```
Comment (radare) [inline libr/debug/trace.c]: what about < 1

File: libr/debug/trace.c [hunk 5]
```diff
@@ -50,24 +50,36 @@ R_API int r_debug_trace_pc (RDebug *dbg, ut64 pc) {
 	RAnalOp op;
 	static ut64 oldpc = 0LL; // Must trace the previously traced instruction
```
Comment (radare) [inline libr/debug/trace.c]: use oldpc = UT64_MAX, 0 is a valid address

File: libr/debug/trace.c [hunk 6]
```diff
@@ -50,24 +50,36 @@ R_API int r_debug_trace_pc (RDebug *dbg, ut64 pc) {
...
+			//eprintf ("trace_pc: cannot read memory at 0x%"PFMT64x"\n", addr);
+			return false;
+		}
+
+		if (r_anal_op (dbg->anal, &op, addr, buf, sizeof (buf)) <= 0) {
+			eprintf ("trace_pc: cannot get opcode size at 0x%"PFMT64x"\n", addr);
+			return false;
+		}
+
+		if (oldpc != 0LL) {
```
Comment (radare) [inline libr/debug/trace.c]: use UT64_MAX instead of 0

**Comment by radare** [comment]
ping

**Comment by radare** [comment]
ping

**Comment by radare** [comment]
please rebase to make it green

**Comment by jduck** [comment]
rebased!

**Comment by radare** [comment]
```
trace.c: In function ‘r_debug_trace_pc’:
trace.c:54:2: error: ‘ri’ undeclared (first use in this function)
trace.c:54:2: note: each undeclared identifier is reported only once for each function it appears in
make[4]: *** [trace.o] Error 1
make[3]: *** [foo] Error 2
make[2]: *** [debug] Error 2
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [all] Error 2
make: *** [all] Error 2
```

**Comment by jduck** [comment]
Trying again...

**Pull request merged**